File: day-01/part-1/badouralix.py
from tool.runners.python import SubmissionPy


class BadouralixSubmission(SubmissionPy):
    def run(self, s):
        """
        :param s: input in string format
        :return: solution flag
        """
        # Parse input and handle the weird edge case where 1010 appears twice
        entries = list(map(int, s.split()))
        if entries.count(1010) >= 2:
            return 1010 * 1010

        # Yolo build a set out of the list of entries
        entries = set(entries)

        # Yolo patch the set in case there is a single 1010 there
        entries.discard(1010)

        # Find elements that sum up to 2020
        for e in entries:
            if (2020 - e) in entries:
                return e * (2020 - e)

File: day-01/part-1/badouralix.js
const { performance } = require('perf_hooks');
const _ = require("lodash")

/**
 * @param {string} s puzzle input in string format
 * @returns {string} solution flag
 */
const run = s => {
  const entries = _.map(_.split(s, "\n"), _.toInteger)

  for (const [i, e] of Object.entries(entries)) {
    for (const f of _.slice(entries, _.toInteger(i) + 1)) {
      if (e + f == 2020) {
        return e * f
      }
    }
  }
};

const start = performance.now();
const answer = run(process.argv[2]);

console.log(`_duration:${performance.now() - start}`);
console.log(answer);
